tests/sentry/tasks/test_derive_code_mappings.py

```diff
@@ -329,6 +329,80 @@ def test_derive_code_mappings_starts_with_app_dot_dot_slash(self):
             assert code_mapping.repository.name == repo_name
 
 
+class TestPhpDeriveCodeMappings(BaseDeriveCodeMappings):
+    def setUp(self):
+        super().setUp()
+        self.platform = "php"
+        self.event_data = self.generate_data(
+            [
+                {"in_app": True, "filename": "/sentry/capybara.php"},
+                {"in_app": True, "filename": "/sentry/potato/kangaroo.php"},
+                {
+                    "in_app": False,
+                    "filename": "/sentry/potato/vendor/sentry/sentry/src/functions.php",
+                },
+            ],
+            self.platform,
+        )
+
+    def test_find_stacktrace_paths_single_project(self):
+        stacktrace_paths = identify_stacktrace_paths(self.event_data)
+        assert set(stacktrace_paths) == {
+            "/sentry/capybara.php",
+            "/sentry/potato/kangaroo.php",
+        }
+
+    def test_handle_duplicate_filenames_in_stacktrace(self):
+        duplicate_data = self.generate_data(
+            [
+                {"in_app": True, "filename": "/sentry/capybara.php"},
+                {"in_app": True, "filename": "/sentry/capybara.php"},
+                {"in_app": True, "filename": "/sentry/potato/kangaroo.php"},
+                {
+                    "in_app": False,
+                    "filename": "/sentry/potato/vendor/sentry/sentry/src/functions.php",
+                },
+            ],
+            self.platform,
+        )
+
+        stacktrace_paths = identify_stacktrace_paths(duplicate_data)
+        assert set(stacktrace_paths) == {
+            "/sentry/capybara.php",
+            "/sentry/potato/kangaroo.php",
+        }
+
+    @responses.activate
+    def test_derive_code_mappings_basic_php(self):
+        repo_name = "php/place"
+        with patch(
+            "sentry.integrations.github.client.GitHubClientMixin.get_trees_for_org"
+        ) as mock_get_trees_for_org:
+            mock_get_trees_for_org.return_value = {
+                repo_name: RepoTree(Repo(repo_name, "master"), ["sentry/potato/kangaroo.php"])
+            }
+            derive_code_mappings(self.project.id, self.event_data)
```

Comment:
How is this working at all if the org does not have `organizations:derive-php-code-mappings` as a feature?

Reply:
This erroneously worked before because the feature flag check existed only inside the `post_process.py`, which is where the celery task to automate code mappings is defined. As shown in this code snippet on line 384, these tests directly called the function where the code mapping logic is defined, `derive_code_mappings`, thus skipping the feature flag check. 

The feature flag check has since been moved to directly inside `derive_code_mappings`, and the tests now fail when the flag is absent, and pass when the flag is present, as expected.